src/main/java/Duke.java

```diff
@@ -1,10 +1,294 @@
+import task.*;
+import storage.Storage;
+import ui.Ui;
+
+import java.util.*;
+
 public class Duke {
+
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    public Duke() {
+
+        this.ui = new Ui();
+        this.storage = new Storage(System.getProperty("user.dir") + "/data/", "duke.txt");
+        this.taskList = storage.load();
+
+    }
+
+    public void run() {
+
+        Scanner sc = new Scanner(System.in);
+
+        ui.printWelcomeMessage();
+
+        boolean isExit = false;
```

Comment:
Would it be better to rename isExit to hasExited or something else? I'm not too sure what it refers to.

Reply:
I wanted to have a boolean instead of a loop but I have already removed this in my recent code.